app/Http/Controllers/Backend/PermissionsController.php

```diff
@@ -0,0 +1,253 @@
+                $request->session()->flash('message', trans('labels.MessageAddRoleOk'));
+            } else {
+                $request->session()->flash('message', trans('labels.MessageAddRoleNotOk'));
+            }
+        } else {
+            $request->session()->flash('message', trans('labels.MessageRoleExists'));
+        }
+        return redirect()->route('admin.permissions.index');
+    }
+
+    /**
+     * Show the form for editing the specified Permissions.
+     *
+     * @param int $id id
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
+    {
+        $modules = $this->getModule();
+        $permission = Permission::findOrFail($id);
+        return view('backend.permissions.edit')->with(['modules' => $modules, 'permission' => $permission]);
+    }
+
+    /**
+     * Update the specified Permissions in storage.
+     *
+     * @param \Illuminate\Http\Request $request request
+     * @param int                      $id      id
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $id)
+    {
+        if (!$this->checkExists($request)) {
+            // get permission with id
+            $role = Permission::findOrFail($id);
+            $role->module = $request->module;
+
+            // set permission
+            $arrRole = $this->setPermission($request);
+            $role->see     = $arrRole['see'];
+            $role->addNew  = $arrRole['addNew'];
+            $role->edit    = $arrRole['edit'];
+            $role->destroy = $arrRole['destroy'];
+            
+            // check save successfull?
+            if ($role->save()) {
+                $request->session()->flash('message', trans('labels.MessageEditRoleOk'));
+            } else {
+                $request->session()->flash('message', trans('labels.MessageEditRoleNotOk'));
+            }
+        } else {
+            $request->session()->flash('message', trans('labels.MessageRoleExists'));
+        }
+        return redirect()->route('admin.permissions.index');
+    }
+
+    /**
+     * Remove the specified Permissions from storage.
+     *
+     * @param int                      $id      id
+     * @param \Illuminate\Http\Request $request request for delete
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy($id, Request $request)
+    {
+        $permission = Permission::findOrFail($id);
+
+        if ($permission->delete()) {
+            $request->session()->flash('message', trans('labels.MessageDelRoleOk'));
+        } else {
+            $request->session()->flash('message', trans('labels.MessageDelRoleNotOk'));
+        }
+        return redirect()->route('admin.permissions.index');
+    }
+
+    /**
+     * Auto get list Module.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    private function getModule()
+    {
+        $modules = [];
+
+        foreach (\Route::getRoutes()->getRoutes() as $route) {
```

Comment:
[rank B] we should avoid to use an expression if for or if block. We should declare and assign it in a variable.